StardewArchipelago/Archipelago/ModsManager.cs

```diff
@@ -70,13 +74,41 @@ public bool IsModStateCorrect(IModHelper modHelper, out string errorMessage)
                 {
                     continue;
                 }
+
                 valid = false;
                 errorMessage += $"{Environment.NewLine}\tMod: {modName}, expected version: {desiredVersion}, current Version: {existingVersion}";
             }
 
             return valid;
         }
 
+        public bool IsPatcherStateCorrect(IModHelper modHelper, out string errorMessage)
+        {
+            var loadedModData = modHelper.ModRegistry.GetAll().ToList();
+            errorMessage = $"The slot you are connecting to requires a content patcher,\r\n mod, but not all expected mods are installed and active.";
+            var valid = true;
+            foreach (var modName in _activeMods)
+            {
+                if (!ModVersions.CPVersions.ContainsKey(modName))
+                {
+                    continue;
+                }
+                var contentName = ModVersions.CPVersions[modName].Keys.First();

```

Comment:
Assuming that you made these a dictionary to account for possible multiple dependencies in the future, why only check the first? Why not check all here?